lesson16/main.js

```diff
@@ -0,0 +1,119 @@
+"use strict";
+
+const jsonUrl = "https://jsondata.okiba.me/v1/json/yFXrE210903000829";
+const newsWrapper = document.getElementById("js-news");
+
+const createElementWithClassName = (element, name) => {
+  const variableName = document.createElement(element);
+  variableName.classList.add(name);
+  return variableName;
+};
+
+//comment-icon
+const commentIcon = createElementWithClassName("span", "comment-icon");
+
+//new-label
+const newLabel = createElementWithClassName("span", "new-label");
+newLabel.textContent = "NEW!";
+
+//news-img
+const imgWrapper = createElementWithClassName("div", "news__img");
+
+
+//tab
+const createNewTab = data => {
+  const tabUl = document.querySelector(".news__tabs");
+  const tabFragment = document.createDocumentFragment();
+
+  data.forEach((key) => {
+    const tabLi = createElementWithClassName("li", "tab");
+    tabLi.textContent = key.category;
+    tabLi.dataset.tab = key.id;
+    if (key.selected === true) {
+      tabLi.classList.add("is-active");
+    }
+
+    tabLi.addEventListener("click", tabSwitch);
+    tabFragment.appendChild(tabLi);
+  });
+
+  tabUl.appendChild(tabFragment);
+};
+
+//news
+const createNewContent = data => {
+  const contentFragment = document.createDocumentFragment();
+  
+  data.forEach( key => {
+    const newsContent = createElementWithClassName("div", "news__content");
+    const contentUl = createElementWithClassName("ul", "news__lists");
+    const titleFragment = document.createDocumentFragment();
+    newsContent.dataset.content = key.id;
+
+    key.article.forEach( array => {
+      const contentLi = createElementWithClassName("li", "news__item");
+      const a = document.createElement("a");
+      a.textContent = array.title;
+      a.href = "#";
+      titleFragment.appendChild(contentLi).appendChild(a);
+    })
+    
+    if (key.selected === true) {
+      newsContent.classList.add("is-show");
+    }
+    contentFragment.appendChild(newsContent).appendChild(contentUl).appendChild(titleFragment);
+  });
+
+  newsWrapper.appendChild(contentFragment);
+};
+
+//tabSwitch
+const tabSwitch = () => {
+  const tabs = document.querySelectorAll("[data-tab]");
+  const contents = document.querySelectorAll("[data-content]");
+
+  const handleClick = e => {
+    e.preventDefault();
+    const targetTab = e.target;
+    const targetTabVal = targetTab.dataset.tab;
+    
+    let i = 0;
+    while (i < tabs.length) {
+      contents[i].classList.remove("is-show");
+      tabs[i].classList.remove("is-active");
+      i++;
+    };
+    
+    newsWrapper.querySelectorAll('[data-content="' + targetTabVal + '"]')[0].classList.add("is-show");
+    targetTab.classList.add("is-active");
+  };
+  
+  let i = 0;
+  while (i < tabs.length) {
+    tabs[i].addEventListener("click", e => handleClick(e));
+    i++;
+  }
+}
```

Comment:
Must
The function is long, so please separate it.

Reply:
For now I implemented lines 80 to 85 of the above code as `hideNewsElements` and lines 87 to 89 as `showNewsElements`.

Should I also separate lines 75 to 95 (as name of like a `addClickEvent`) like this? ↓

```
const tabSwitch = () => {
  const tabs = document.querySelectorAll("[data-tab]");
  const contents = document.querySelectorAll("[data-content]");

  addClickEvent(tabs, contents);
};

const addClickEvent = (tabs, contents) => {
  const handleClick = (e) => {
    const targetTab = e.target;
    const targetTabVal = targetTab.dataset.tab;
    e.preventDefault();
    hideNewsElements(tabs, contents);
    showNewsElements(targetTab, targetTabVal);
  };

  let i = 0;
  while (i < tabs.length) {
    tabs[i].addEventListener("click", (e) => handleClick(e));
    i++;
  }
};
```
I thought it might make it a little harder to read.

Please kindly give me an advice. Thank you.